BDArmory/Control/MissileFire.cs

```diff
@@ -6331,7 +6331,7 @@ void GuardMode()
 
                                 if (firedMissiles < maxMissilesOnTarget)
                                 {
-                                    if (CurrentMissile.TargetingMode == MissileBase.TargetingModes.Radar && _radarsEnabled && !CurrentMissile.radarLOAL && MaxradarLocks < multiMissileTgtNum)
+                                    if (CurrentMissile.TargetingMode == MissileBase.TargetingModes.Radar && _radarsEnabled && !CurrentMissile.radarLOAL && (MaxradarLocks < vesselRadarData.GetLockedTargets().Count() + 1))
```

Comment:
This should really be using `Count`, not `Count()`: https://stackoverflow.com/questions/4098186/lists-count-vs-count

Also, is the logic correct here? You can fire more radar guided missiles than the max number of locks by disabling the radar or if they're LOAL?

Reply:
Changed to use Count and also <= instead of <.
Regarding the conditions, the radar enabled check is probably redundant (not sure, maybe there's some purpose there that I'm not seeing either?) since guard mode can't fire radar missiles without the radar on. That being said, the radarLOAL condition I'm more confused by. I think it's legacy code but it's to prevent SARH missiles from being fired when there's not enough available locks, but radarLOAL doesn't define SARH missiles. I think this ought to read
`targetDistance < CurrentMissile.activeRadarRange` since that's what defines if the missile is being fired as an SARH missiles, including if the missile is a mixed SARH/ARH missile but I wasn't entirely sure. I think we should probably discuss this before merging the pull request.

Reply:
| since guard mode can't fire radar missiles without the radar on.
This changed with the unguided missiles refactor a little while ago.
The current ``if (CurrentMissile.TargetingMode == MissileBase.TargetingModes.Radar && _radarsEnabled && !CurrentMissile.radarLOAL && MaxradarLocks <= vesselRadarData.GetLockedTargets().Count)`` looks correct and appropriate; the check here is for 1) radar being on; 2) missile is slaved to launcher radar; 3) there are available locks.
The _radarsEnabled check, if false, instead has the AI treat the radar missile as dumbfire ordinance later in the code (and to prevent firing pure SARH missiles while the radar is off due to an incoming HARM). The LOAL check is the correct one here; yes, the range that radar missiles will switch to onboard guidance is activeRadarRange, but missiles fired without a radar lock will only attempt to acquire one after launch if radarLOAL = true; dumbfiring a !radarLOAL missile with an activeRadarRange > 0 will just be a big, expensive rocket, so we want to avoid doing that unnecessarily.

Reply:
OK, this should be good to merge then.